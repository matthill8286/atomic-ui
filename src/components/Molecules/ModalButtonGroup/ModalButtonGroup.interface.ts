import { ButtonProps } from '@/components/Atoms/Button'

export type ModalButtonGroupAlignment = 'space-between' | 'right' | 'center'

export interface ModalButtonProps extends ButtonProps {
  onClick?: () => void
  buttonLabel?: string | React.ReactNode
  href?: string
}

export interface ModalButtonGroupProps {
  className?: string
  key?: string
  buttonAlignment?: ModalButtonGroupAlignment
  buttonWidth?: number
  primaryButtonProps?: ModalButtonProps
  secondaryButtonProps?: ModalButtonProps
  showButtonSeparator?: boolean
}
