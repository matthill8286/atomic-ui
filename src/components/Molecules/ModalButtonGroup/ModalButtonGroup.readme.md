# Modal Button Group molecule

This components represents a two or a single button in a context of a modal. It can be accessed directly by using the `Modal`-component props or used directly within a `Modal`.
