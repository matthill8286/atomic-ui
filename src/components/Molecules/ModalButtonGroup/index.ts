export * from './ModalButtonGroup'
export * from './ModalButtonGroup.interface'
