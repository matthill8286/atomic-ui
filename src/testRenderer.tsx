import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
