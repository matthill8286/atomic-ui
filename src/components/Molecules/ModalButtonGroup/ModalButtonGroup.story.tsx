import { action } from '@storybook/addon-actions'
import { boolean, select } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { ModalButtonGroup } from './ModalButtonGroup'
import { ModalButtonGroupAlignment, ModalButtonGroupProps } from './ModalButtonGroup.interface'
import Readme from './ModalButtonGroup.readme.md'

export const types: ModalButtonGroupAlignment[] = ['space-between', 'right', 'center']

const knobs = (): Partial<ModalButtonGroupProps> => {
  return {
    buttonAlignment: select('Button type', types, types[0]),
    showButtonSeparator: boolean('With Button Seperator', true),
  }
}

storiesOf('Design System/Molecules/Modals/ModalButtonGroup', module)
  .add(
    'Default',
    () => {
      return (
        <ModalButtonGroup
          {...knobs()}
          primaryButtonProps={{
            buttonLabel: 'Call to action',
            actionType: 'primary',
            onClick: action('onClick call to action'),
          }}
          secondaryButtonProps={{
            buttonLabel: 'Close',
            actionType: 'outlined',
            onClick: action('onClick close'),
          }}
        />
      )
    },
    { info: Readme }
  )
  .add(
    'Primary Button control',
    () => {
      return (
        <ModalButtonGroup
          {...knobs()}
          buttonAlignment="center"
          primaryButtonProps={{
            buttonLabel: 'Call to action',
            actionType: 'primary',
            onClick: action('onClick call to action'),
          }}
        />
      )
    },
    { info: Readme }
  )
