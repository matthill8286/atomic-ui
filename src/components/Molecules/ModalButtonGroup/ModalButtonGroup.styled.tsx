import { Button } from '@/components/Atoms/Button'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { ModalButtonGroupProps } from './ModalButtonGroup.interface'

export const StyledModalFooter = styled.div<ModalButtonGroupProps>`
  ${({ showButtonSeparator, theme, buttonAlignment: buttonType }) => css`
    border-top: 1px solid ${showButtonSeparator ? theme.color.grey2 : `transparent`};
    padding-top: ${({ theme }) => theme.spacing.base.lg};
    width: 100%;
    display: flex;
    flex-direction: row;

    ${buttonType === 'space-between' &&
      `
      justify-content: space-between;
    `}

    ${buttonType === 'right' &&
      `
      justify-content: flex-end;
    `}

    ${buttonType === 'center' &&
      `
      justify-content: center;
      width: 85%;
      margin: 0 auto;
    `}

    ${media.maxSm} {
      flex-direction: column-reverse;
    }
  `}
`

export const StyledModalFooterPrimary = styled.div<ModalButtonGroupProps>`
  ${({ buttonAlignment: buttonType }) =>
    buttonType === 'space-between' &&
    `
    text-align: right;
    `}
  ${({ buttonAlignment: buttonType }) =>
    buttonType === 'right' &&
    `
    text-align: right;
    `}
    ${({ buttonAlignment: buttonType, buttonWidth, theme }) =>
      buttonType === 'center'
        ? `
      margin: 0 auto 0 ${theme.spacing.base.xs};
      width: ${typeof buttonWidth === 'number' ? `${buttonWidth}px ` : '100%'}
    `
        : `
      margin: 0 ${theme.spacing.base.md}
    `}
`
export const StyledModalFooterSecondary = styled.div<ModalButtonGroupProps>`
  ${({ theme, buttonAlignment: buttonType }) => css`
    ${buttonType === 'right' &&
      `
      margin-right: ${theme.spacing.base.md};
    `}

    ${
      buttonType === 'center'
        ? `
      margin: 0 ${theme.spacing.base.xs} 0 auto;
       width: 100%;
    `
        : `
      margin: 0 ${theme.spacing.base.md}
    `
    }

    ${media.maxSm} {
      margin-top: ${({ theme }) => theme.spacing.base.sm};
      margin-right: 0;
    }
  `}
`

export const StyledModalFooterButton = styled(Button)`
  width: 100%;
  margin: 0 auto;
  max-width: 400px;
  ${media.maxSm} {
    width: 100%;
  }
`
