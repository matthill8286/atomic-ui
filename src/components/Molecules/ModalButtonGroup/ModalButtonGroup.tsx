import * as React from 'react'
import { ModalButtonGroupProps } from './ModalButtonGroup.interface'
import {
  StyledModalFooter,
  StyledModalFooterButton,
  StyledModalFooterPrimary,
  StyledModalFooterSecondary,
} from './ModalButtonGroup.styled'

export const ModalButtonGroup: React.FC<ModalButtonGroupProps> = ({
  buttonAlignment = 'space-between',
  primaryButtonProps,
  secondaryButtonProps,
  buttonWidth,
  showButtonSeparator = true,
}): JSX.Element => {
  if (!primaryButtonProps && !secondaryButtonProps) {
    return <React.Fragment />
  }

  const handleClick = (method?: () => void) => (ev?: React.MouseEvent) => {
    if (ev) ev.stopPropagation()
    if (method) method()
  }

  return (
    <StyledModalFooter
      showButtonSeparator={showButtonSeparator}
      buttonAlignment={buttonAlignment}
      data-test="filtered-modal-footer-buttons">
      {secondaryButtonProps && (
        <StyledModalFooterSecondary buttonAlignment={buttonAlignment}>
          <StyledModalFooterButton
            actionType="secondary"
            data-test={'filtered-secondary-modal-footer-buttons'}
            onClick={
              secondaryButtonProps.onClick ? handleClick(secondaryButtonProps.onClick) : undefined
            }
            {...secondaryButtonProps}>
            {secondaryButtonProps.buttonLabel}
          </StyledModalFooterButton>
        </StyledModalFooterSecondary>
      )}
      {primaryButtonProps && (
        <StyledModalFooterPrimary buttonAlignment={buttonAlignment} buttonWidth={buttonWidth}>
          <StyledModalFooterButton
            actionType="primary"
            fullWidth={!!buttonWidth}
            data-test="filtered-primary-modal-footer-buttons"
            onClick={
              primaryButtonProps.onClick ? handleClick(primaryButtonProps.onClick) : undefined
            }
            {...primaryButtonProps}>
            {primaryButtonProps.buttonLabel}
          </StyledModalFooterButton>
        </StyledModalFooterPrimary>
      )}
    </StyledModalFooter>
  )
}

ModalButtonGroup.displayName = 'ModalButtonGroup'
