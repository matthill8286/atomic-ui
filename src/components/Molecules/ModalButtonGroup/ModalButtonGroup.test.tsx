import * as React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { ModalButtonGroup } from './ModalButtonGroup'
import { ModalButtonGroupProps } from './ModalButtonGroup.interface'

describe('ModalButtonGroup', () => {
  const props: Partial<ModalButtonGroupProps> = {
    primaryButtonProps: {
      buttonLabel: 'Call to action',
      actionType: 'primary',
      onClick: jest.fn(),
    },
    secondaryButtonProps: {
      buttonLabel: 'Bookmark',
      actionType: 'outlined',
      onClick: jest.fn(),
    },
  }

  it('Should render a ModalButtonGroup', () => {
    const tree = renderWithTheme(<ModalButtonGroup {...props} />)
    expect(tree).toMatchSnapshot()
  })
  it('Should render with alignment.', () => {
    const tree = renderWithTheme(<ModalButtonGroup buttonAlignment="right" {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('Should not render with no Button', () => {
    const tree = renderWithTheme(<ModalButtonGroup />)
    expect(tree).toMatchSnapshot()
  })
})
