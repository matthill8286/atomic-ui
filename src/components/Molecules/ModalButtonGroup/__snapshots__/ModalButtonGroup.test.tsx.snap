// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButtonGroup Should not render with no Button 1`] = `null`;

exports[`ModalButtonGroup Should render a ModalButtonGroup 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #00C8DC;
  background: #FFFFFF;
  color: #00C8DC;
}

.c2 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c2:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c2 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
  color: #FFFFFF;
}

.c5 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c5:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c5 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c0 {
  border-top: 1px solid #E5E5E5;
  padding-top: 32px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  text-align: right;
  margin: 0 24px;
}

.c1 margin:0 24px @media (max-width:512px) {
  margin-top: 16px;
  margin-right: 0;
}

.c3 {
  width: 100%;
  margin: 0 auto;
  max-width: 400px;
}

@media (max-width:512px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (max-width:512px) {
  .c3 {
    width: 100%;
  }
}

<div
  className="c0"
  data-test="saiyan-modal-footer-buttons"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      data-test="saiyan-secondary-modal-footer-buttons"
      disabled={false}
      onClick={[Function]}
      size="md"
      type="button"
    >
      Bookmark
    </button>
  </div>
  <div
    className="c4"
  >
    <button
      className="c5 c3"
      data-test="saiyan-primary-modal-footer-buttons"
      disabled={false}
      onClick={[Function]}
      size="md"
      type="button"
    >
      Call to action
    </button>
  </div>
</div>
`;

exports[`ModalButtonGroup Should render with alignment. 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #00C8DC;
  background: #FFFFFF;
  color: #00C8DC;
}

.c2 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c2:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c2 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  background: #00C8DC;
  border: 2px solid #00C8DC;
  color: #FFFFFF;
  color: #FFFFFF;
}

.c5 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c5:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c5 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c0 {
  border-top: 1px solid #E5E5E5;
  padding-top: 32px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c4 {
  text-align: right;
  margin: 0 24px;
}

.c1 {
  margin-right: 24px;
}

.c1 margin:0 24px @media (max-width:512px) {
  margin-top: 16px;
  margin-right: 0;
}

.c3 {
  width: 100%;
  margin: 0 auto;
  max-width: 400px;
}

@media (max-width:512px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (max-width:512px) {
  .c3 {
    width: 100%;
  }
}

<div
  className="c0"
  data-test="saiyan-modal-footer-buttons"
>
  <div
    className="c1"
  >
    <button
      className="c2 c3"
      data-test="saiyan-secondary-modal-footer-buttons"
      disabled={false}
      onClick={[Function]}
      size="md"
      type="button"
    >
      Bookmark
    </button>
  </div>
  <div
    className="c4"
  >
    <button
      className="c5 c3"
      data-test="saiyan-primary-modal-footer-buttons"
      disabled={false}
      onClick={[Function]}
      size="md"
      type="button"
    >
      Call to action
    </button>
  </div>
</div>
`;
